src/mmw/apps/modeling/mapshed/calcs.py

```diff
@@ -250,8 +250,11 @@ def ag_ls_c_p(geom):
 
         ag_lscp = namedtuple('Ag_LS_C_P',
                              [col[0] for col in cursor.description])
-
-        return ag_lscp(*(cursor.fetchone()))
+        ag_results = { k: v or 0.0 for (k, v) in
```

Comment:
The same can be done here:

```diff
diff --git a/src/mmw/apps/modeling/mapshed/calcs.py b/src/mmw/apps/modeling/mapshed/calcs.py
index 659c927..9d35937 100644
--- a/src/mmw/apps/modeling/mapshed/calcs.py
+++ b/src/mmw/apps/modeling/mapshed/calcs.py
@@ -232,16 +232,16 @@ def ag_ls_c_p(geom):
           ), clipped_counties_with_area AS (
               SELECT ST_Area(geom_clipped) /
                      ST_Area(ST_SetSRID(ST_GeomFromText(%s),
-                                        4326)) AS clip_percent,
+                                        4326)) AS clip_pct,
                      clipped_counties.*
               FROM clipped_counties
           )
-          SELECT SUM(hp_ls * clip_percent) AS hp_ls,
-                 SUM(hp_c * clip_percent) AS hp_c,
-                 SUM(hp_p * clip_percent) AS hp_p,
-                 SUM(crop_ls * clip_percent) AS crop_ls,
-                 SUM(crop_c * clip_percent) AS crop_c,
-                 SUM(crop_p * clip_percent) AS crop_p
+          SELECT COALESCE(SUM(hp_ls * clip_pct), 0.0) AS hp_ls,
+                 COALESCE(SUM(hp_c * clip_pct), 0.0) AS hp_c,
+                 COALESCE(SUM(hp_p * clip_pct), 0.0) AS hp_p,
+                 COALESCE(SUM(crop_ls * clip_pct), 0.0) AS crop_ls,
+                 COALESCE(SUM(crop_c * clip_pct), 0.0) AS crop_c,
+                 COALESCE(SUM(crop_p * clip_pct), 0.0) AS crop_p
           FROM clipped_counties_with_area;
           '''
 
```

`clip_percent` was changed to `clip_pct` here for consistency with the last query.

Doing this in SQL would save all the conversion and reiteration happening here in Python.